week-3/Homework/mandatory/5-journey-planner.js

```diff
@@ -1,25 +1,154 @@
+  break down the whole task into smaller steps that make our job easier.
+
+  This technic is also referred as problem decomposition. It helps you to reduce scope of the problem 
+  by only focusing on a small chunk of the whole problem at a time.)
+*/
+
+/*
+  Implement the function getTransportModes that
+   - Accepts an array containing the location and available transport modes
+     e.g: ["Tower Bridge", "tube", "river boat"]
+   - Returns an array including the available transport modes to the given location
+     e.g: ["tube", "river boat"]
+  
+  Hint: Use the corresponding array method to split the array.
 */
+function getTransportModes(transport) {
+  var possibleRoads = transport.slice(1,3)
+  return possibleRoads;
+ }
 
-function journeyPlanner() {
+/*
+  Implement the function isAccessibleByTransportMode that
+   - Accepts two parameters:
+     1) First parameter is an array of transport modes
+        e.g: ["tube", "river boat"]
+     2) Second parameter is a string containing a transport mode
+        e.g: "river boat"
+     
+   - Returns 
+     * True if the location in the first parameter is accessible by the transport mode given in second parameter
+     * Otherwise, returns false
 
+  Hint: Use the corresponding array method to decide if an element is member of an array.
+*/
+function isAccessibleByTransportMode(mode, strMode) { 
+  if (mode.includes(strMode)){
+    return true 
+  } else {
+    return false
+  }
+}
+
+/*
+  Implement the function getLocationName that
+   - Accepts a location and available transports in an array
+     e.g:["Tower Bridge", "tube", "river boat"]
+   - Returns the name of the location
+     e.g: "Tower Bridge"
+*/
+function getLocationName(location) {
+  return location[0]
+ }
+
+/*
+ We arrived at the final method. it won't take long if you use the previously implemented functions wisely.
+
+ Finish up the implemention of the function journeyPlanner that
+  - Accepts two parameters:
+    1) An array with a list of locations' and their transports
+       e.g:
+       [
+         ["Angel", "tube", "bus"],
+         ["London Bridge", "tube", "river boat"]
+       ]
+    2) A string containing a transport mode
+       e.g: "bus"
+
+  - Returns an array of where I can go if I only want to use a specific mode of transport.
+    NOTE: only the location names should be returned, not the name of transports.
+
+  Hint:
+   - Use the function you implemented above.
+   - Use array method to remove locations that is not accessible by the given transportMode.
+   - Use array method to manipulate its elements.
+   
+  Advanced challange: try to use arrow function when invoking an array method.
+*/
+function journeyPlanner(locations, transportMode) {
+  var availableTransports = locations.filter(a=>a.includes(transportMode))
+  // Implement the function body
+  return availableTransports.map(getLocationName)
 }
```

Comment:
This function is wrong. what you are looking for is if the locations have the transport mode then return the locations name which has transport mode that we are looking for.

Reply:
I did return the location names which have the transport mode  by using the   function getLocationName from the previous exercise as required in the instruction so this will only show the first name of the array which is the location name